core/lib/dal/src/contract_verification_dal.rs

```diff
@@ -278,6 +365,32 @@ impl ContractVerificationDal<'_, '_> {
         .execute(&mut transaction)
         .await?;
 
+        // Add a new etherscan verification request only if the source code is
+        // a solc single file or a standard json input. Others are not supported by
+        // the Etherscan API.
+        if let SourceCodeData::SolSingleFile(_) | SourceCodeData::StandardJsonInput(_) =
+            verification_info.request.req.source_code_data
+        {
```

Comment:
Should maybe this logic be moved into a separate method which would only be invoked if etherscan is configured?
Otherwise, for chains with no etherscan we will be inserting this data for no reason.

Reply:
My initial idea was to be able to turn off the verifier but still collect the requests for further processing. But as I think about it more, I agree that it's better to persist requests only if the verifier is enabled. Changed.